pkg/services/ngalert/api/api_provisioning.go

```diff
@@ -390,9 +390,24 @@ func (srv *ProvisioningSrv) RouteGetAlertRuleGroup(c *contextmodel.ReqContext, f
 
 // RouteGetAlertRulesExport retrieves all alert rules in a format compatible with file provisioning.
 func (srv *ProvisioningSrv) RouteGetAlertRulesExport(c *contextmodel.ReqContext) response.Response {
-	groupsWithTitle, err := srv.alertRules.GetAlertGroupsWithFolderTitle(c.Req.Context(), c.OrgID)
-	if err != nil {
-		return ErrResp(http.StatusInternalServerError, err, "failed to get alert rules")
+	folderUID := c.Query("folder_uid")
+	group := c.Query("group")
+	var groupsWithTitle []alerting_models.AlertRuleGroupWithFolderTitle
+	if group != "" {
+		if folderUID == "" {
+			return ErrResp(http.StatusBadRequest, nil, "group name must be specified together with folder_uid parameter")
+		}
+		groupWithFolderTitle, err := srv.alertRules.GetAlertRuleGroupWithFolderTitle(c.Req.Context(), c.OrgID, folderUID, group)
+		if err != nil {
+			return ErrResp(http.StatusInternalServerError, err, "failed to get alert rules")
+		}
+		groupsWithTitle = []alerting_models.AlertRuleGroupWithFolderTitle{groupWithFolderTitle}
```

Comment:
Maybe just delegate this to the existing method. That way it covers error cases for `ErrAlertRuleGroupNotFound` and automatically stays in sync the the path-based endpoint to some degree.
```suggestion
		return srv.RouteGetAlertRuleGroupExport(c, folderUID, group)
```